### PR TITLE
Lesson title update

### PR DESCRIPTION
**Pull request by nunibaranes**
- Changing styles according to the design: https://projects.invisionapp.com/d/main?origin=v7#/console/20017367/427640412/preview?scrollOffset=4002
- Listens to keyboardEvent: 
`escape` should close edit mode and revert last changes
`enter` should close edit mode and save last changes
- on blur, the input should close edit mode and save last changes
- Adds error state while the the title is empty on edit mode
- Prevent setting an empty title for a lesson and revert the last changes.
- Adds auto focus to a new lesson.
- Add transitions to hidden elements to create a smooth transition when they appear

Notes:
I didn't remove the delete button (if it removed the user can't delete the lesson)

### CODE REVIEW

File: src/components/Screen/CourseEditorScreen/LessonHeader/style.module.scss [hunk 1]
```diff
@@ -59,12 +60,42 @@
...
+      outline: 1px dashed transparent;
+      transition: outline 0.3s;
+
+      .lesson-title-text {
+        display: inline-block;
+        white-space: nowrap;
+        text-overflow: ellipsis;
+        overflow: hidden;
+        flex-grow: 1;
+        width: 100%;
```
Comment (dvira) [inline src/components/Screen/CourseEditorScreen/LessonHeader/style.module.scss:80]: You could use mixin here.

File: src/components/Screen/CourseEditorScreen/LessonHeader/EditbaleLessonTitle.tsx [hunk 2]
```diff
@@ -10,54 +17,90 @@ import Icon, { IconType } from '../../../common/Icon';
...
+        </div>
         <LessonDeleteButton lesson={lesson} showInput={showInput} />
       </div>
       <div className={cc([classes['input-wrapper'], { [classes['hidden']]: !showInput }])}>
         <WMInput
           className={cc([classes.input])}
           ref={inputTitle}
           value={inputValue}
-          onChange={onChange}
+          onChange={(e) => setInputValue(e.target.value)}
```
Comment (dvira) [inline src/components/Screen/CourseEditorScreen/LessonHeader/EditbaleLessonTitle.tsx:99]: Please wrap all callbacks passed to our components with `useCallback`.

Comment (dr15) [inline src/components/Screen/CourseEditorScreen/LessonHeader/EditbaleLessonTitle.tsx:99]: I would change this to a function wrapped in `useCallback()`

File: src/components/Screen/CourseEditorScreen/LessonHeader/EditbaleLessonTitle.tsx [hunk 3]
```diff
@@ -10,54 +17,90 @@ import Icon, { IconType } from '../../../common/Icon';
...
     if (!inputTitle.current) return;
 
-    setInputValue(lesson?.title ?? '');
+    setInputValue(lesson?.title);
     inputTitle.current.focus();
     setShowInput(true);
+  }, [setInputValue, setShowInput, lesson?.title]);
+
+  // onApprove - save the inputValue, and hide the input
+  const onApprove = () => {
```
Comment (dr15) [inline src/components/Screen/CourseEditorScreen/LessonHeader/EditbaleLessonTitle.tsx:43]: I would wrap this and all the other functions after it with `useCallback()`

**Review by dr15 — CHANGES_REQUESTED** [review]
A couple of tiny changes

**Review by dvira — CHANGES_REQUESTED** [review]

**Review by dvira — APPROVED** [review]
FIRE!!!!

**Pull request merged**